aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -169,6 +179,126 @@ def get(  # type: ignore[override]
 
         return super().get(name, max_age, transform, force_fetch, **sdk_options)
 
+    @overload
+    def set(
+        self,
+        name: str,
+        value: list[str],
+        *,
+        overwrite: bool = False,
+        description: str = "",
+        parameter_type: Literal["StringList"] = "StringList",
+        tier: Literal["Standard", "Advanced", "Intelligent-Tiering"] = "Standard",
+        kms_key_id: str | None = "None",
+        **sdk_options,
+    ): ...
+
+    @overload
+    def set(
+        self,
+        name: str,
+        value: str,
+        *,
+        overwrite: bool = False,
+        description: str = "",
+        parameter_type: Literal["SecureString"] = "SecureString",
+        tier: Literal["Standard", "Advanced", "Intelligent-Tiering"] = "Standard",
+        kms_key_id: str,
+        **sdk_options,
+    ): ...
+
+    @overload
+    def set(
+        self,
+        name: str,
+        value: str,
+        *,
+        overwrite: bool = False,
+        description: str = "",
+        parameter_type: Literal["String"] = "String",
+        tier: Literal["Standard", "Advanced", "Intelligent-Tiering"] = "Standard",
+        kms_key_id: str | None = None,
+        **sdk_options,
+    ): ...
+
+    def set(
+        self,
+        name: str,
+        value: str | list[str],
+        *,
+        overwrite: bool = False,
+        description: str = "",
+        parameter_type: SSM_PARAMETER_TYPES = "String",
+        tier: SSM_PARAMETER_TIER = "Standard",
+        kms_key_id: str | None = None,
+        **sdk_options,
+    ) -> PutParameterResponse:
+        """
+        Sets a parameter in AWS Systems Manager Parameter Store.
+
+        Parameters
+        ----------
+        name: str
+            The fully qualified name includes the complete hierarchy of the parameter name and name.
+        value: str
+            The parameter value
+        parameter_type: str, optional
+            Type of the parameter.  Allowed values are String, StringList, and SecureString
+        overwrite: bool, optional
+            If the parameter value should be overwritten, False by default
+        tier: str, optional
+            The parameter tier to use.  Allowed values are Standard, Advanced, and Intelligent-Tiering
+        description: str, optional
+            The description of the parameter
+        kms_key_id: str, optional
+            The KMS key id to use to encrypt the parameter
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the Parameter Store get_parameter API call
```

Comment:
Same order as declared on the signature:

```suggestion
        name: str
            The fully qualified name includes the complete hierarchy of the parameter name and name.
        value: str
            The parameter value
        overwrite: bool, optional
            If the parameter value should be overwritten, False by default
        description: str, optional
            The description of the parameter
        parameter_type: str, optional
            Type of the parameter.  Allowed values are String, StringList, and SecureString
        tier: str, optional
            The parameter tier to use.  Allowed values are Standard, Advanced, and Intelligent-Tiering
        kms_key_id: str, optional
            The KMS key id to use to encrypt the parameter
        sdk_options: dict, optional
            Dictionary of options that will be passed to the Parameter Store get_parameter API call
```